addons/point_of_sale/static/src/js/models.js

```diff
@@ -221,7 +221,13 @@ exports.PosModel = Backbone.Model.extend({
         fields: ['id', 'journal_ids','name','user_id','config_id','start_at','stop_at','sequence_number','login_number'],
         domain: function(self){ return [['state','=','opened'],['user_id','=',session.uid]]; },
         loaded: function(self,pos_sessions){
-            self.pos_session = pos_sessions[0]; 
+            self.pos_session = pos_sessions[0];
+
+            // reset the cashier to the current user if session is new
+            if (self.db.load('pos_session_id') !== self.pos_session.id) {
+                self.set_cashier(self.user);
+            }
```

Comment:
I would try to move logic like this to the `get_cashier` function. This way developers can use that function without having to worry about this session stuff.